File: Assets/Plugins/Android/src/com/playphone/multinet/unity/MNDirectButtonUnity.java
package com.playphone.multinet.unity;

import com.playphone.multinet.*;
import com.unity3d.player.UnityPlayer;

public class MNDirectButtonUnity {
    public static boolean isVisible() {
        MNUnity.MARK();

        return MNDirectButton.isVisible();
    }

    public static boolean isHidden() {
        MNUnity.MARK();

        return MNDirectButton.isHidden();
    }

    public static void initWithLocation(final int param1_int) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectButton.initWithLocation(param1_int);
            }
        });
    }

    public static void show() {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectButton.show();
            }
        });
    }

    public static void hide() {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectButton.hide();
            }
        });
    }

}

File: Assets/Plugins/iOS/PlayPhoneMultiNet/unity/MNDirectButtonUnity.m
//
//  MNDirectButtonUnity.m
//  PlayPhone SDK for Unity
//
//  Copyright (c) 2012 PlayPhone Inc. All rights reserved.
//

#import "MNDirectButton.h"

#import "MNDirectButtonUnity.h"

EXTERN_C
bool _MNDirectButton_IsVisible() {
    MARK;

    return [MNDirectButton isVisible];
}

EXTERN_C
bool _MNDirectButton_IsHidden() {
    MARK;

    return [MNDirectButton isHidden];
}

EXTERN_C
void _MNDirectButton_InitWithLocation(int location) {
    MARK;

    [MNDirectButton initWithLocation:location];
}

EXTERN_C
void _MNDirectButton_Show() {
    MARK;

    [MNDirectButton show];
}

EXTERN_C
void _MNDirectButton_Hide() {
    MARK;

    [MNDirectButton hide];
}


